src/RangeSlider.elm

```diff
@@ -0,0 +1,349 @@
+            100 / (commonAttributes.max - commonAttributes.min)
+
+        value =
+            clamp commonAttributes.min commonAttributes.max valueAttributes.value
+    in
+    { left = 0.0, right = (commonAttributes.max - value) * progressRatio }
+
+
+sliderInputView : CommonAttributes a -> ValueAttributes a -> Html a
+sliderInputView commonAttributes valueAttributes =
+    let
+        track =
+            [ Html.Attributes.class "input-range__track", onClick (onOutsideRangeClick commonAttributes) ]
+
+        progressPercentages =
+            calculateProgressPercentages commonAttributes valueAttributes
+
+        progressAttributes =
+            [ Html.Attributes.class "input-range__progress"
+            , Html.Attributes.style "left" <| String.fromFloat progressPercentages.left ++ "%"
+            , Html.Attributes.style "right" <| String.fromFloat progressPercentages.right ++ "%"
+            , onClick (onInsideRangeClick commonAttributes valueAttributes)
+            ]
+    in
+    div [ Html.Attributes.class "input-range-container" ]
+        [ Html.input
+            [ Html.Attributes.type_ "range"
+            , Html.Attributes.min <| String.fromFloat commonAttributes.min
+            , Html.Attributes.max <| String.fromFloat commonAttributes.max
+            , Html.Attributes.step <| String.fromFloat commonAttributes.step
+            , Html.Attributes.value <| String.fromFloat valueAttributes.value
+            , Html.Attributes.class "input-range"
+            , onChange valueAttributes.change
+            , onInput valueAttributes.input
+            ]
+            []
+        , div track []
+        , div progressAttributes []
+        ]
+
+
+sliderLabelView : CommonAttributes a -> ValueAttributes a -> Html a
+sliderLabelView commonAttributes valueAttributes =
+    div
+        [ Html.Attributes.class "input-range-labels-container" ]
+        [ div
+            [ Html.Attributes.class "input-range-label" ]
+            [ Html.text <| commonAttributes.minFormatter { value = commonAttributes.min } ]
+        , div
+            [ Html.Attributes.class "input-range-label input-range-label--current-value" ]
+            [ Html.text <| valueAttributes.formatter { value = valueAttributes.value, max = commonAttributes.max } ]
+        , div
+            [ Html.Attributes.class "input-range-label" ]
+            [ Html.text <| commonAttributes.maxFormatter { value = commonAttributes.max } ]
+        ]
+
+
+
+-- API
+
+
+defaultFormatter : { value : Float } -> String
+defaultFormatter value =
+    String.fromFloat value.value
+
+
+defaultValueFormatter : { value : Float, max : Float } -> String
+defaultValueFormatter values =
+    if values.value == values.max then
+        ""
+
+    else
+        String.fromFloat values.value
+
+
+initSingleSlider :
+    { min : Float
+    , max : Float
+    , step : Float
+    , value : Float
+    , onChange : String -> a
+    , onInput : String -> a
+    , onClick : String -> a
```

Comment:
Any strong reason to use String here? All our values are Float maybe we could keep it consistent here. :)